modules/extensions/galasa-extensions-parent/dev.galasa.ras.couchdb/src/test/java/dev/galasa/ras/couchdb/internal/CouchdbDirectoryServiceTest.java

```diff
@@ -645,4 +645,55 @@ public void testGetRunsByRunNameWithBadlyFormattedJsonResponseThrowsCorrectError
             "Invalid JSON response returned from CouchDB"
         );
     }
+
+    //------------------------------------------
+    //
+    // Tests for getting runs by group
+    //
+    //------------------------------------------
+
+    @Test
+    public void testGetRunsByGroupReturnsRunOk() throws Exception {
+        // Given...
+        String run1Id = "run1-id";
+        String run1Name = "ABC123";
+        String groupName = "testGroup";
+        String urlEncodedGroupName = URLEncoder.encode('"' + groupName + '"', StandardCharsets.UTF_8);
+        TestStructureCouchdb mockRun1 = createRunTestStructure(run1Id, run1Name, groupName);
+
+        ViewResponse mockViewResponse = new ViewResponse();
+        ViewRow run1Row = new ViewRow();
+        run1Row.id = run1Id;
+        run1Row.key = run1Name;
+        run1Row.doc = mockRun1;
+
+        List<ViewRow> mockRows = new ArrayList<>();
+        mockRows.add(run1Row);
+        mockViewResponse.rows = mockRows;
+
+        String baseUri = "http://my.uri";
+        String groupNameViewUri = baseUri + "/" + CouchdbRasStore.RUNS_DB + "/_design/docs/_view/" + CouchdbRasStore.RUN_GROUP_VIEW_NAME;
+
+        List<HttpInteraction> interactions = List.of(
+            // Get the run document
+            new GetRunsFromCouchdbViewInteraction(groupNameViewUri + "?key=" + urlEncodedGroupName + "&include_docs=true", HttpStatus.SC_OK, mockViewResponse)
+        );
+
+        MockLogFactory mockLogFactory = new MockLogFactory();
+        MockAsyncCloseableHttpClient httpClient = new MockAsyncCloseableHttpClient(interactions);
```

Comment:
Minor - I don't think we need to use a mock async client in these tests (looks like the existing getRunsByRunName tests also don't need to use it - oops!)